tests/network/vmi_istio.go

```diff
@@ -0,0 +1,310 @@
+			Selector: map[string]string{
+				"app": selector,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port: svcDeclaredTestPort,
+				},
+			},
+		},
+	}
+}
+
+func newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(appLabel string, network *v1.Network, ports []v1.Port) *v1.VirtualMachineInstance {
+	vmi := libvmi.NewCirros(
+		libvmi.WithNetwork(network),
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(ports...)),
+		libvmi.WithAppLabel(appLabel),
+		withIstioAnnotation(),
+	)
+	// Istio-proxy requires service account token to be mounted
+	tests.AddServiceAccountDisk(vmi, "default")
+	return vmi
+}
+
+var _ = SIGDescribe("[Serial] Istio", func() {
+	BeforeEach(func() {
+		if !istioServiceMeshDeployed() {
+			Skip("Istio service mesh is required for service-mesh tests to run")
+		}
+	})
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+	network := &v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+
+	Context("Virtual Machine with masquerade interface and explicitly specified ports", func() {
+		verifyClientServerConnectivity := func(serverVMI *v1.VirtualMachineInstance, targetPort int) (*batchv1.Job, error) {
+			By("Starting HTTP Server")
+			tests.StartHTTPServer(vmi, targetPort)
+
+			By("Getting back the VMI IP")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			vmiIP := vmi.Status.Interfaces[0].IP
+
+			By("Running job to send a request to the server")
+			return virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Create(
+				context.Background(),
+				generateBatchJobSpec(vmiIP, fmt.Sprintf("%d", targetPort), tests.NewHelloWorldJobHTTP),
+				metav1.CreateOptions{},
+			)
+		}
+		BeforeEach(func() {
+			libnet.AddLabelToNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel, "enabled")
+			ports := []v1.Port{
+				{Port: svcDeclaredTestPort},
+				{Port: svcUndeclaredTestPort},
+			}
+			vmi = newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(
+				vmiAppSelector,
+				network,
+				ports,
+			)
+			service = newServiceWithAppSelector(
+				fmt.Sprintf("%s-service", vmi.Name),
+				vmi.Namespace,
+				vmiAppSelector,
+			)
+
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			By("Creating VMI and associated Service")
+			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		})
+		AfterEach(func() {
+			libnet.RemoveLabelFromNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel)
```

Comment:
`tests.BeforeTestCleanup()` should  remove vms and services.

Reply:
Done